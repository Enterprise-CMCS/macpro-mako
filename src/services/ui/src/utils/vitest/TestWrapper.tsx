import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ReactLocation, Router } from "@tanstack/react-location";

const queryClient = new QueryClient();
const location = new ReactLocation();

export const TestWrapper = ({ children }: any) => (
  <QueryClientProvider client={queryClient}>
    <Router location={location} routes={[]}>
      {children}
    </Router>
  </QueryClientProvider>
);
