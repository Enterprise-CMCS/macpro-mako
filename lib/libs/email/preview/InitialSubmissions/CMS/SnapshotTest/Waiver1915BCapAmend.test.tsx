import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import CMSWaiver1915bCapitatedAmendment from "../Waiver1915bCapitated/Amendment";

describe("Initial Submission CMS Email Snapshot Test", () => {
  it("renders a Amendment Waiver Capitated Preview Template", () => {
    const { asFragment } = render(<CMSWaiver1915bCapitatedAmendment />);

    expect(asFragment()).toMatchSnapshot();
  });
});
