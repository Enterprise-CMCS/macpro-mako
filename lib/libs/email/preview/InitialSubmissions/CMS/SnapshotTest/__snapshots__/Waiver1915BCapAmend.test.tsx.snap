// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Initial Submission CMS Email Snapshot Test > renders a Amendment Waiver Capitated Preview Template 1`] = `
<DocumentFragment>
  <meta
    content="text/html; charset=UTF-8"
    http-equiv="Content-Type"
  />
  <meta
    name="x-apple-disable-message-reformatting"
  />
  <div
    style="display: none; overflow: hidden; line-height: 1px; opacity: 0; max-height: 0; max-width: 0;"
  >
    1915(b) Amendment Submitted
    <div>
       ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿
    </div>
  </div>
  <table
    align="center"
    border="0"
    cellpadding="0"
    cellspacing="0"
    role="presentation"
    style="max-width: 37.5em; background-color: rgb(245, 245, 245);"
    width="100%"
  >
    <tbody>
      <tr
        style="width: 100%;"
      >
        <td>
          <table
            align="center"
            border="0"
            cellpadding="0"
            cellspacing="0"
            role="presentation"
            style="background-color: rgb(0, 113, 189); padding: 16px;"
            width="100%"
          >
            <tbody>
              <tr>
                <td>
                  <a
                    href="https://mako-dev.cms.gov/"
                    style="color: rgb(6, 125, 247); text-decoration: none; display: block; max-width: 112px;"
                    target="_blank"
                  >
                    <img
                      alt="OneMAC Logo"
                      height="40"
                      src="https://mako-dev.cms.gov/onemac-logo.png"
                      style="max-width: 112px;"
                      width="112"
                    />
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            style="margin: 0px 8px 8px 8px; padding: 8px;"
          >
            <h1
              style="color: rgb(51, 51, 51); font-size: 20px; font-weight: bold; margin-bottom: 15px;"
            >
              The OneMAC Submission Portal received a 1915(b) Amendment waiver submission:
            </h1>
            <ul
              style="margin-left: -20px;"
            >
              <li>
                <p
                  style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                >
                  The submission can be accessed in the OneMAC application, which you can find at 
                  <a
                    href="https://mako-dev.cms.gov/"
                    style="color: rgb(6, 125, 247); text-decoration: none;"
                    target="_blank"
                  >
                    this link
                  </a>
                  .
                </p>
              </li>
              <li>
                <p
                  style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                >
                  If you are not already logged in, please click the "Login" link at the top of the page and log in using your Enterprise User Administration (EUA) credentials.
                </p>
              </li>
              <li>
                <p
                  style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                >
                  After you have logged in, you will be taken to the OneMAC application. The submission will be listed on the dashboard page, and you can view its details by clicking on its ID number.
                </p>
              </li>
            </ul>
            <table
              align="center"
              border="0"
              cellpadding="0"
              cellspacing="0"
              role="presentation"
              width="100%"
            >
              <tbody>
                <tr>
                  <td>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              State or Territory:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              CO
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Name:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              George Harrison
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Email Address:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              george@example.com
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Amendment Waiver Number:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              CO-1234.R21.01
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Waiver Authority:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              1915(b)
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Proposed Effective Date:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              March 2, 2023
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <hr
                      style="width: 100%; border-top: 1px solid #0071BD; margin: 16px 0px;"
                    />
                    <p
                      style="font-size: 14px; line-height: 24px; margin: .5em;"
                    />
                    <h2
                      style="font-size: 16px;"
                    >
                      Summary:
                    </h2>
                    <p />
                    <p
                      style="font-size: 14px; line-height: 1.4; margin: 12px 0px; color: rgb(51, 51, 51);"
                    >
                      This submission includes necessary documentation for requested updates to the state’s Medicaid plan, in alignment with CMS requirements.
                    </p>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        />
                      </tbody>
                    </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <hr
              style="width: 100%; border-top: 1px solid #0071BD; margin: 16px 0px;"
            />
            <h2
              style="font-size: 16px;"
            >
              Files:
            </h2>
            <table
              align="center"
              border="0"
              cellpadding="0"
              cellspacing="0"
              role="presentation"
              style="margin-bottom: 2px; margin-top: 2px;"
              width="100%"
            >
              <tbody
                style="width: 100%;"
              >
                <tr
                  style="width: 100%;"
                >
                  <td
                    align="left"
                    data-id="__react-email-column"
                    style="width: 50%; vertical-align: top;"
                  >
                     
                    <span>
                      <p
                        style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                      >
                        1915(b) Comprehensive (Capitated) Waiver Cost Effectiveness Spreadsheets:
                      </p>
                       
                    </span>
                  </td>
                  <td
                    data-id="__react-email-column"
                    style="vertical-align: top;"
                  >
                    <p
                      style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                    >
                      <span>
                        capitated-waiver-cost-effectiveness-spreadsheet.pdf
                      </span>
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
            <table
              align="center"
              border="0"
              cellpadding="0"
              cellspacing="0"
              role="presentation"
              style="margin-bottom: 2px; margin-top: 2px;"
              width="100%"
            >
              <tbody
                style="width: 100%;"
              >
                <tr
                  style="width: 100%;"
                >
                  <td
                    align="left"
                    data-id="__react-email-column"
                    style="width: 50%; vertical-align: top;"
                  >
                     
                    <span>
                      <p
                        style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                      >
                        1915(b) Comprehensive (Capitated) Waiver Application Pre-print:
                      </p>
                       
                    </span>
                  </td>
                  <td
                    data-id="__react-email-column"
                    style="vertical-align: top;"
                  >
                    <p
                      style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                    >
                      <span>
                        capitated-waiver-application.pdf
                      </span>
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
            <p
              style="font-size: 14px; line-height: 1.4; margin: 12px 0px; color: rgb(51, 51, 51);"
            >
              Thank you.
            </p>
          </div>
          <table
            align="center"
            border="0"
            cellpadding="0"
            cellspacing="0"
            role="presentation"
            style="font-size: 14px; padding: 0px 0px; font-weight: 300; background-color: rgb(0, 113, 189); color: rgb(255, 255, 255); text-align: center;"
            width="100%"
          >
            <tbody>
              <tr>
                <td>
                  <table
                    align="center"
                    border="0"
                    cellpadding="0"
                    cellspacing="0"
                    role="presentation"
                    style="font-size: 14px; padding: 0px 0px; font-weight: 300; background-color: rgb(0, 113, 189); color: rgb(255, 255, 255); text-align: center;"
                    width="100%"
                  >
                    <tbody>
                      <tr>
                        <td>
                          <p
                            style="font-size: 14px; line-height: 24px; margin: 8px; color: rgb(255, 255, 255);"
                          >
                            U.S. Centers for Medicare & Medicaid Services
                          </p>
                          <p
                            style="font-size: 14px; line-height: 24px; margin: 8px; color: rgb(255, 255, 255);"
                          >
                            © 2023 | 7500 Security Boulevard, Baltimore, MD 21244
                          </p>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
